ios/RN/RNCamera.m

```diff
@@ -421,40 +403,110 @@ - (void)record:(NSDictionary *)options resolve:(RCTPromiseResolveBlock)resolve r
+{
+    [self findPrimaryFace:sampleBuffer];
+
+    if (self.canAppendBuffer) {
+        if (self.videoWriter.status != AVAssetWriterStatusWriting) {
+            [self.videoWriter startWriting];
+            [self.videoWriter startSessionAtSourceTime:CMSampleBufferGetPresentationTimeStamp(sampleBuffer)];
+            dispatch_async(dispatch_get_main_queue(), ^{
+                self.timer = [NSTimer scheduledTimerWithTimeInterval:self.maxDuration target:self selector:@selector(stopAssetWriter) userInfo:nil repeats:NO];
+            });
+        }
+        [self.writerInput appendSampleBuffer:sampleBuffer];
+    }
+}
+
+-(void)findPrimaryFace:(CMSampleBufferRef)sampleBuffer {
+    CVPixelBufferRef pixelBuffer = CMSampleBufferGetImageBuffer(sampleBuffer);
+    CIImage *image = [CIImage imageWithCVPixelBuffer:pixelBuffer];
+    CIImage *orientedImage = [image imageByApplyingCGOrientation:kCGImagePropertyOrientationUpMirrored];
+
+    VNDetectFaceRectanglesRequest *faceDetectionReq = [VNDetectFaceRectanglesRequest new];
+    NSDictionary *d = [[NSDictionary alloc] init];
+    VNImageRequestHandler *handler = [[VNImageRequestHandler alloc] initWithCIImage:orientedImage options:d];
+    [handler performRequests:@[faceDetectionReq] error:nil];
+
+    VNFaceObservation *primaryFace;
+    CGPoint primaryFaceCenter;
+    float primaryFaceSize;
+
+    for(VNFaceObservation *observation in faceDetectionReq.results){
+        if(observation){
+            float size = observation.boundingBox.size.height * observation.boundingBox.size.width;
+            if (!primaryFace || size > primaryFaceSize) {
+                primaryFace = observation;
+                primaryFaceCenter = CGPointMake(CGRectGetMidX(observation.boundingBox), CGRectGetMidY(observation.boundingBox));
+                primaryFaceSize = size;
+            }
+        }
+    }
+
+    if ([faceDetectionReq.results count] && !self.exposureTimeout) {
+        [self setExposure:primaryFaceCenter];
+        self.exposureTimeout = YES;
+        dispatch_async(dispatch_get_main_queue(), ^{
+            self.exposureTimer = [NSTimer scheduledTimerWithTimeInterval:1.0 target:self selector:@selector(resetExposureTimeout) userInfo:nil repeats:NO];
+        });
+    }
+}
+
+- (void)resetExposureTimeout;
+{
+    self.exposureTimeout = NO;
+}
+
+- (void)setExposure:(CGPoint) point;
 {
-    [self.movieFileOutput stopRecording];
+    AVCaptureDevice *device = [self.videoCaptureDeviceInput device];
+    [device lockForConfiguration:nil];
+    CGPoint scaledPoint = CGPointMake(point.x * self.layer.bounds.size.width, (1-point.y) * self.layer.bounds.size.height);
+    CGPoint devicePoint = [self.previewLayer captureDevicePointOfInterestForPoint:scaledPoint];
+    [device setExposurePointOfInterest:devicePoint];
+    if ([device isExposureModeSupported:AVCaptureExposureModeContinuousAutoExposure])
```

Comment:
Do we need to set the exposure and white balance mode to continuous every time we change the point of interest or can we just do this on setup?

Reply:
Yeah, this not necessary. Moved exposure into a separate function that is called once when the session is initialized.